org.adempiere.base/src/org/compiere/db/PreparedStatementProxy.java

```diff
@@ -95,6 +98,12 @@ protected void init() {
 			p_stmt = conn.prepareStatement(p_vo.getSql(), p_vo
 					.getResultSetType(), p_vo.getResultSetConcurrency());
 		} catch (Exception e) {
+			if (localConn && m_conn != null) {
+				try {
+					m_conn.close();
+					m_conn = null;
+				} catch (SQLException e1) {}
+			}

```

Comment:
doesn't need the localConn stuff, just need to close this.close() here.

Reply:
Hi @hengsin 
this.close() is not available at this point
the "localconn" is to avoid closing a connection that is managed elsewhere, I just need to close here the connections created locally in the same method.

Reply:
ok, you need to change StatementProxy.close to protected. Calling the close method of StatementProxy should be the correct approach here, not the method above.

Reply:
Thanks @hengsin - tested your suggestion and it worked fine.